clang/lib/Lex/PPDirectives.cpp

```diff
@@ -177,6 +180,20 @@ static bool isLanguageDefinedBuiltin(const SourceManager &SourceMgr,
   return false;
 }
 
+static bool isReservedCXXAttributeName(Preprocessor &PP, IdentifierInfo *II) {
+  const LangOptions &Lang = PP.getLangOpts();
+  if (Lang.CPlusPlus &&
+      hasAttribute(AttributeCommonInfo::Syntax::AS_CXX11, /*Scope*/ nullptr, II,
+                   PP.getTargetInfo(), Lang) > 0) {
+    AttributeCommonInfo::Kind AttrKind = AttributeCommonInfo::getParsedKind(
+        II, /*Scope*/ nullptr, AttributeCommonInfo::Syntax::AS_CXX11);
+    return !((AttrKind == AttributeCommonInfo::Kind::AT_Likely ||
```

Comment:
I'm kinda confused how this works, can you explain it here?  Why the likely/unlikely exception here?

Reply:
The names `likely()` and `unlikely()` are not reserved, as noted [here](https://github.com/llvm/llvm-project/pull/106036#discussion_r1730709436). Therefore, using these names with `()` should not trigger a warning.

```cpp

#define likely 1 // expected-warning {{likely is a reserved attribute identifier}}
#undef likely    // expected-warning {{likely is a reserved attribute identifier}}

#define unlikely 1 // expected-warning {{unlikely is a reserved attribute identifier}}
#undef unlikely    // expected-warning {{unlikely is a reserved attribute identifier}}

#define likely()   // ok
#define unlikely() // ok
```

Reply:
Got it, thanks!  Forgot about that/didn't recall it.